### PR TITLE
More robust flake8 linting

### PR DESCRIPTION
**Pull request by zaneselvans**
This PR changes a bunch of our string formatting to use the Python 3.6 and later f-strings (I used the automated `flynt` tool to convert most of them), adds an RST docstring linter to the flake8 configuration, and consolidates a bunch of configuration for the linters into a single place.  It also adds all the flake8 linter plugins to the `pre-commit` environment for flake8 so that they actually get run by the pre-commit hook.

Note that the RST and f-string linters don't currently show up in Atom, because of a bug in the Atom flake8 linter plugin which fails to recognize error codes with more than one letter in the prefix.

Also updated the README and some other documentation files, but that's a work in progress.

### CODE REVIEW

File: src/pudl/constants.py [hunk 1]
```diff
@@ -245,7 +245,7 @@
     'no. 2  oil', 'blended oil', 'no 2. oil', '# 6 oil', 'nno. 2 oil',
     '#2 fuel', 'oill', 'oils', 'gas/oil', 'no.2 oil gas', '#2 fuel oil',
     'oli', 'oil (#6)', 'oil/diesel', '2 oil', '#6 hvy oil', 'jet fuel',
-    'diesel/compos', 'oil-8', 'oil {6}', 'oil-unit #1', 'bbl.', 'oil.',
+    'diesel/compos', 'oil-8', 'oil {6}', 'oil-unit #1', 'bbl.', 'oil.',  # noqa: FS003
```
Comment (cmgosnell) [inline src/pudl/constants.py:248]: why? that is the issue here? is it the {'s?

File: src/pudl/etl.py [hunk 2]
```diff
@@ -445,9 +445,8 @@ def _etl_epacems(etl_params, datapkg_dir, pudl_settings):
                                 datapkg_dir=datapkg_dir)
         epacems_tables.append(list(transformed_df_dict.keys())[0])
     if logger.isEnabledFor(logging.INFO):
-        time_message = "    Loading    EPA CEMS took {}".format(
-            time.strftime("%H:%M:%S",
-                          time.gmtime(time.monotonic() - start_time)))
+        delta_t = time.strftime("%H:%M:%S", time.gmtime(time.monotonic() - start_time))
+        time_message = f"    Loading    EPA CEMS took {delta_t}"
```
Comment (cmgosnell) [inline src/pudl/etl.py:449]: nbd, but there are some weird extra spaces in here

File: src/pudl/workspace/datastore.py [hunk 3]
```diff
@@ -50,15 +50,15 @@ class Datastore:
     def __init__(self, pudl_in, loglevel="WARNING", verbose=False,
                  sandbox=False, timeout=7):
         """
-        Datastore manages file retrieval for PUDL datasets.
+        Datastore `manages` file retrieval for PUDL datasets.
```
Comment (cmgosnell) [inline src/pudl/workspace/datastore.py:53]: ?

**Review by cmgosnell — COMMENTED** [review]
👍🏻

File: src/pudl/constants.py (hunk 1 above)
Comment (zaneselvans) [reply src/pudl/constants.py:248]: Yes, the `flake8-use-fstring` linter misinterprets the `"oil {6}"` string literal as an f-string without an f-prefix.

File: src/pudl/etl.py (hunk 2 above)
Comment (zaneselvans) [reply src/pudl/etl.py:449]: They're in the original too.

File: src/pudl/workspace/datastore.py (hunk 3 above)
Comment (zaneselvans) [reply src/pudl/workspace/datastore.py:53]: No idea wtf is gonig on there.

File: src/pudl/etl.py (hunk 2 above)
Comment (cmgosnell) [reply src/pudl/etl.py:449]: okay. but should we remove it?

File: src/pudl/constants.py (hunk 1 above)
Comment (cmgosnell) [reply src/pudl/constants.py:248]: got it! thanks

File: src/pudl/workspace/datastore.py (hunk 3 above)
Comment (cmgosnell) [reply src/pudl/workspace/datastore.py]: should we note `manages` this? manages seems fine unless this is actually referring to something.

**Review by cmgosnell — APPROVED** [review]

File: src/pudl/etl.py (hunk 2 above)
Comment (zaneselvans) [reply src/pudl/etl.py]: Okay, it's gone. I think it must have been vestigal from before the generator was used for CEMS and there was more sequential progressive output.

**Pull request merged**